mesh/attribute_sets.hpp

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) 2010-2023, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_ATTRIBUTE_SETS
+#define MFEM_ATTRIBUTE_SETS
+
+#include "../config/config.hpp"
+#include "../general/arrays_by_name.hpp"
+
+#include <iostream>
+#include <map>
+#include <set>
+#include <string>
+
+namespace mfem
+{
+
+class AttributeSets
+{
+public:
+   /// Named sets of element attributes
+   ArraysByName<int> attr_sets;
+
+   /// Named sets of boundary attributes
+   ArraysByName<int> bdr_attr_sets;
+
+   AttributeSets() {}
+
+   /// @brief Create a copy of the internal data to the provided @a copy.
+   void Copy(AttributeSets &copy) const;
```

Comment:
Can remove this in favor of copy/move constructors and `operator=` (as in `ArraysByName<T>`).